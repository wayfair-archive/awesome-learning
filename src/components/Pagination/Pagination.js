import React from "react";
import { Link } from "gatsby";
import { PAGINATION } from "../../constants";
import "./Pagination.scss";

const Pagination = ({
  prevPagePath,
  nextPagePath,
  hasNextPage,
  hasPrevPage
}) => (
  <div className="Pagination">
    {hasPrevPage && (
      <div className="Pagination-prev">
        <Link rel="prev" to={prevPagePath} className="Pagination-prev-link">
          {PAGINATION.PREV_PAGE}
        </Link>
      </div>
    )}
    {hasNextPage && (
      <div className="Pagination-next">
        <Link rel="next" to={nextPagePath} className="Pagination-next-link">
          {PAGINATION.NEXT_PAGE}
        </Link>
      </div>
    )}
  </div>
);

export default Pagination;
