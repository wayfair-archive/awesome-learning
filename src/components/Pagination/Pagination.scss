@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.Pagination {
  @include margin-top(2);
  display: flex;
  &-prev {
    justify-content: flex-start;
    width: 100%;
    text-align: left;

    &-link {
      color: $color--secondary;
      font-size: 26px;
      font-weight: bold;

      &:hover,
      &:focus {
        color: $color--primary;
      }

    }

  }

  &-next {
    justify-content: flex-end;
    width: 100%;
    text-align: right;

    &-link {
      color: $color--secondary;
      font-size: 26px;
      font-weight: bold;

      &:hover,
      &:focus {
        color: $color--primary;
      }
    }

  }

}