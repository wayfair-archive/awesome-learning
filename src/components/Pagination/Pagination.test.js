import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';

describe('Pagination', () => {
  const props = {
    prevPagePath: '/page/1',
    nextPagePath: '/page/3',
    hasNextPage: true,
    hasPrevPage: true
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Pagination {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
