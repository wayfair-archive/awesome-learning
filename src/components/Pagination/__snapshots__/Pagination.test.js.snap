// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div
  className="Pagination"
>
  <div
    className="Pagination-prev"
  >
    <Link
      className="Pagination-prev-link"
      rel="prev"
      to="/page/1"
    >
      ← PREV
    </Link>
  </div>
  <div
    className="Pagination-next"
  >
    <Link
      className="Pagination-next-link"
      rel="next"
      to="/page/3"
    >
      → NEXT
    </Link>
  </div>
</div>
`;
